### ISSUE TITLE
CMS still spins up ImageMagick although ImageOptimize is installed and configured to use Imgix?

### ISSUE THREAD

**Issue by thijskaspers** [nystudio107/craft3-imageoptimize#373]
### Question
After updating to Craft 4.4.5 (we came from the latest 4.3.~) our site was extremely slow. We noticed that all image transformations had to be recreated, possibly because of the addition of the "allow upscale" setting at the image transforms settings.

The strange thing is though: we have ImageOptimize set-up to use Imgix so the handling of image transforms won't hit our server performance wise, but somehow it still did. We've noticed that the site has been slow all of a sudden, and ImageMagick was running with high CPU/memory to handle image transforms. After the pages finally load, the images on the page are correctly coming from Imgix...

What am I missing here, why would ImageMagick spin up, while we use Imgix as the transform method?

Perhaps the CMS also stores assets locally, so both Imgix and ImageMagick are being used?

Hopefully someone can point us in the right direction

### Additional context
Craft v4.4.5 / ImageOptimize v4.0.4

**Comment by khalwat** [comment]
I'm thinking this may be a bug or regression in the changes to the way they are handling images in Craft CMS 4.4.

Will investigate.

**Comment by khalwat** [comment]
Is there a chance you're using the Async Queue plugin?

**Comment by khalwat** [comment]
https://github.com/ostark/craft-async-queue/issues/57

**Comment by thijskaspers** [comment]
Hey Andrew, we don't use the Async Queue plugin. We do have `runQueueAutomatically` set to false, and use a cronjob (docker container with Ofelia) to run `craft queue/run` every minute

**Comment by andersaloof** [comment]
Hi,

I have had a similiar challenge, however we were on latest v3 then upgraded to 4.4.4. After upgrading we ended up with over 100k "transform" jobs in the queue, even if they are should be served from ImgIx.

Did some debugging and seems to me the reason lies here:
https://github.com/craftcms/cms/blob/4.4/src/elements/Asset.php#L1824

The _url() function ends up calling https://github.com/craftcms/cms/blob/4.4/src/elements/Asset.php#L1897 which in turns calls https://github.com/craftcms/cms/blob/4.4/src/imagetransforms/ImageTransformer.php#L98 since the transform didn't exist before, it queues a transform job, then later in the getUrl() function it overrides the URL from the plugin so that the image ends up being loaded from ImgIx after all.

**Comment by apoupart** [comment]
Hello.

We have the same issue since the latest Craft update in one of the website I'm working on.
Our host provider call us because we generate more than 200gb worth of tmp file in a few hours. just like andersaloof, we ended up with a few thousand jobs in the queue.

As of right now, to stop generating an infinite amount of task and tmp file, we ended up temporarily deactiving the "Image Optimize" plugin.

**Comment by khalwat** [comment]
Filed an issue here: https://github.com/craftcms/cms/issues/13018

**Comment by khalwat** [comment]
Addressed in: https://github.com/nystudio107/craft-imageoptimize/commit/bef35adde1cbedaf24733b8de3231b58c58e5317

**Important**: You will need to be running Craft 4.4.7 or later for this fix to work (at the time of this writing, it isn't out yet, but it will be in the next release)

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-imageoptimize": "dev-develop-v4 as 4.0.5”,
```
Then do a `composer clear-cache && composer update`

**Comment by thijskaspers** [comment]
Thanks @khalwat!

**Comment by thijskaspers** [comment]
Any idea when this fix will be released? :)

**Comment by khalwat** [comment]
Just tagged & released 4.0.5
